novawallet/Common/Services/Web3AlertService/PushNotificationsService.swift

```diff
@@ -0,0 +1,141 @@
+import Foundation
+import UserNotifications
+import UIKit
+import FirebaseMessaging
+import RobinHood
+import SoraKeystore
+import SoraFoundation
+
+enum PushNotificationsStatus {
+    case authorized
+    case on
+    case denied
+    case notDetermined
+}
+
+protocol PushNotificationsServiceProtocol {
+    var statusObservable: Observable<PushNotificationsStatus?> { get set }
+    func setup()
+    func register(
+        completionQueue: DispatchQueue?,
+        completion: @escaping (PushNotificationsStatus) -> Void
+    )
+    func updateStatus()
+}
+
+final class PushNotificationsService: NSObject, PushNotificationsServiceProtocol {
```

Comment:
As I remember we have new version of this service called PushNotificationStatusService